File: backend/app/models/test_run.py
from dataclasses import dataclass
from datetime import datetime

from app.models.status import TestStatus


@dataclass
class TestRun:
    run_id: str
    project_id: str
    test_suite_id: str
    software_version: str
    status: TestStatus
    started_at: datetime
    completed_at: datetime | None
    created_at: datetime

File: frontend/src/app/features/test-runs/models/test-run.ts
import { TestStatus } from '../../../shared/models/test-status';

export interface TestRun {
  id: string;
  run_id: string;
  project_id: string;
  test_suite_id: string;
  software_version: string;
  status: TestStatus;
  started_at: string;
  completed_at: string | null;
  created_at: string;
}

export interface CreateTestRunRequest {
  project_id: string;
  test_suite_id: string;
  software_version: string;
}

export interface TestRunFilters {
  project_id?: string;
  status?: TestStatus;
  software_version?: string;
  date_from?: string;
  date_to?: string;
}